typescript/cli/src/commands/core.ts

```diff
@@ -43,22 +56,45 @@ export const config: CommandModuleWithContext<{
       describe: 'Create an advanced ISM & hook configuration',
       default: false,
     },
-    ismOut: outputFileCommandOption('./configs/ism.yaml'),
-    hooksOut: outputFileCommandOption('./configs/hooks.yaml'),
+    config: outputFileCommandOption(
+      './configs/core-config.yaml',
+      false,
+      'The path to a JSON or YAML file with a core deployment config.',
+    ),
   },
-  handler: async ({ context, ismAdvanced, ismOut, hooksOut }) => {
+  handler: async ({ context, ismAdvanced, config }) => {
     logGray('Hyperlane Core Configure');
     logGray('------------------------');
 
-    await createIsmConfigMap({
+    // Create default Ism config (advanced or trusted)
+    let defaultIsm: IsmConfig;
+    if (ismAdvanced) {
+      logBlue('Creating a new advanced ISM config');
+      logBoldUnderlinedRed('WARNING: USE AT YOUR RISK.');
+      logRed(
+        'Advanced ISM configs require knowledge of different ISM types and how they work together topologically. If possible, use the basic ISM configs are recommended.',
+      );
+      defaultIsm = await createIsmConfig(context);
+    } else {
+      defaultIsm = await createTrustedRelayerConfig(context);
+    }
+
+    // Create default and required Hook config
+    const defaultHook = await createHookConfig(
       context,
-      outPath: ismOut,
-      shouldUseDefault: !ismAdvanced,
-    });
-    await createHooksConfigMap({
+      'Select default hook type',
+    );
+    const requiredHook = await createHookConfig(
       context,
-      outPath: hooksOut,
-    });
+      'Select required hook type',
+    );
+
+    // Validate
+    IsmConfigSchema.parse(defaultIsm);
+    HookConfigSchema.parse(requiredHook);
+    HookConfigSchema.parse(defaultHook);
```

Comment:
any reason not to parse with the CoreConfigSchema?
maybe you can put the hook parse inside of the createHookConfig fn and ditto for createIsmConfig

Reply:
Will investigate and fix in https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3880